protocol/avdl/keybase1/simple_fs.avdl

```diff
@@ -0,0 +1,170 @@
+@namespace("keybase.1")
+
+protocol SimpleFS {
+
+  import idl "common.avdl";
+
+/*
+  File manipulation interface
+  
+  Caller provides OpID to one of these methods to begin an operation:
+    list
+    listRecursive
+    copy
+    open
+    move
+    remove
+  then calls one of the following until until status is no longer pending
+  or operation is cancelled:
+    readList (after list or listRecursive)
+    read (after open)
+    write (after open)
+    check (after copy, move or remove)
+  Caller can optionally block by calling wait()
+  Operation must be closed by calling close
+ 
+*/
+
+  // Random GUID provided by caller 
+  fixed OpID(16);
+  
+  enum DirentType {
+    FILE,
+    DIR,
+    SYM,
+    EXEC
+  }
+
+  record Dirent {
+    string path;
+    Time time;
+    int size;   // size of file or count of dir entries (TBD - mmay be expensive/unnecessary)
+    DirentType direntType;
+  }
+
+  @typedef("int")
+  record ErrorNum {}
+
+  enum Status {
+    finished_0,
+    pending_1,
+    invalidated_2,
+    error_3
+  }
+
+  enum openFlags {
+    READ_0,         // exclusive of other flags
+    REPLACE_1,      // overwrite
+    EXISTING_2,     // fail if not present already
+    WRITE_4,        // read is implied if not write, executable, or directory
+    EXECUTABLE_8,
+    DIRECTORY_16
+  }
+
+  @typedef("int")
+  record Progress {}
+
+  record ListResult {
+    array<Item> items;
+    Progress progress;
+  }
+
+  record FileContent {
+    bytes data
+    Progress progress;
+  }
+
+  record OpDescription {
+    OpID opID,
+    string command,     // list, listRecursive, read, write, copy, move, remove
+    string src,         // source or path
+    string dest         // destination if copy or move
+  }
+
+  /**
+   Begin list if items in directory at path
+   Retrieve results with getList()
+   Can be a single file to get flags/status
+   */
+  void list(OpID opID, string path);
+
+  /**
+   Begin recursive list if items in directory at path
+   */
+  void listRecursive(OpID opID, string path);
+
+  /**
+   Get list if Dirents in progress. Can indicate status of pending
+   to get more entries.
+   */
+  ListResult readList(OpID opID);
+
+  /**
+   Begin copy of file or directory 
+   */
+  void copy(OpID opID, string src, string dest);
+
+  /**
+   Begin recursive copy of directory 
+   */
+  void copyRecursive(OpID opID, string src, string dest);
+
+  /**
+   Begin move of file or directory, from/to KBFS only
+   */
+  void move(OpID opID, string src, string dest);
+
+  /**
+   Rename file or directory, KBFS side only
+   */
+  void rename(string src, string dest);
```

Comment:
maybe we should have a type for paths that are kbfsonly or not. it can be a typedef of string but so we can know what's what, like `kbfspath` or `kbfsOrLocalPath` or something